dvc/commands/studio.py

```diff
@@ -0,0 +1,213 @@
+        return device_code, token_uri
+
+    def save_config(self, hostname, token):
+        with self.config.edit("global") as conf:
+            conf["studio"]["token"] = token
+            conf["studio"]["url"] = hostname
+
+
+class CmdStudioLogout(CmdBase):
+    def run(self):
+        from dvc.ui import ui
+
+        with self.config.edit("global") as conf:
+            if not get_in(conf, ["studio", "token"]):
+                ui.error_write("Not logged in to Studio.")
+                return 1
+
+            del conf["studio"]["token"]
+
+        ui.write("Logged out from Studio")
+        return 0
+
+
+class CmdStudioToken(CmdBase):
+    def run(self):
+        from dvc.ui import ui
+
+        conf = self.config.read("global")
+        token = get_in(conf, ["studio", "token"])
+        if not token:
+            ui.error_write("Not logged in to Studio.")
+            return 1
+
+        ui.write(token)
+        return 0
+
+
+def add_parser(subparsers, parent_parser):
+    STUDIO_HELP = "Authenticate dvc with Iterative Studio"
+    STUDIO_DESCRIPTION = (
+        "Authorize dvc with Studio and set the token. When this is\n"
+        "set, DVC uses this to share live experiments and notify\n"
+        "Studio about pushed experiments."
+    )
+
+    studio_parser = subparsers.add_parser(
+        "studio",
+        parents=[parent_parser],
+        description=append_doc_link(STUDIO_DESCRIPTION, "studio"),
+        help=STUDIO_HELP,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    studio_subparser = studio_parser.add_subparsers(
+        dest="cmd",
+        help="Use `dvc studio CMD --help` to display command-specific help.",
+    )
+    fix_subparsers(studio_subparser)
+
+    STUDIO_LOGIN_HELP = "Authenticate DVC with Studio host"
+    STUDIO_LOGIN_DESCRIPTION = (
+        "By default, this command authorize dvc with Studio with\n"
+        " default scopes and a random  name as token name."
+    )
+    login_parser = studio_subparser.add_parser(
+        "login",
+        parents=[parent_parser],
+        description=append_doc_link(STUDIO_LOGIN_DESCRIPTION, "studio/login"),
+        help=STUDIO_LOGIN_HELP,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+
+    login_parser.add_argument(
+        "-H",
+        "--hostname",
+        action="store",
+        default=None,
+        help="The hostname of the Studio instance to authenticate with.",
+    )
+    login_parser.add_argument(
+        "-s",
+        "--scopes",
+        action="store",
+        default=None,
+        help="The scopes for the authentication token. ",
+    )
+
+    login_parser.add_argument(
+        "-n",
+        "--name",
+        action="store",
+        default=None,
+        help="The name of the authentication token. It will be used to\n"
+        "identify token shown in Studio profile.",
+    )
+
+    login_parser.add_argument(
+        "-d",
+        "--use-device-code",
```

Comment:
What does this flag mean? Aren't we using device code like thing even if we don't open a link ourself?

Reply:
It is something similar to what is implemented at https://learn.microsoft.com/en-us/cli/azure/authenticate-azure-cli-interactively

Reply:
Looks like `azure login` has two different login mechanism, default being Auth Code Flow, and other being Device Code Flow. `--use-device-code` is being used to change that default. Since we only have one type of flow, it does not seem like we need this?

Reply:
Can we change this to `--no-open`/`--no-browser`/`--no-web`, etc?

Reply:
I think `--use-device-code` should be fine since this implies user wants to use the device code itself and it matches the behavior of existing tools like `az login`

Reply:
User has to use device code, no matter what. Again, `az` does not use device code by default and the option is to force that.

Reply:
Updated.